src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,42 @@
 
 class Animal {
   // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(food) {
+    if (food instanceof Carnivore) {
+      return;
+    }
+
+    const index = Animal.alive.indexOf(food);
+
+    if (!food.hidden) {
+      food.health -= 50;
+    }
+
+    if (food.health <= 0) {
+      Animal.alive.splice(index, 1);
```

Comment:
u can make it, using filter and checking how many health does it have & it's instancof